bluemira/geometry/tools.py

```diff
@@ -280,10 +304,13 @@ def interpolate_bspline(
         a bluemira wire that contains the bspline
     """
     points = Coordinates(points).T
-    return BluemiraWire(
-        cadapi.interpolate_bspline(points, closed, start_tangent, end_tangent),
-        label=label,
-    )
+    try:
```

Comment:
I think the decorator would have a better UX if you passed the error into the decorator (in this case `FreeCADError`) then you wouldnt need to modify the function to use the decorator.

eg
```python
@fallback_to(_make_polygon_fallback, cadapi.FreeCADError)
...

Reply:
Right but could we still log upon failure and then fallback with this approach?

Reply:
Actually perhaps we could, let me have a play!